syntax = "proto3";
option go_package = "/protocol/base";


message ClientMsg {
    bytes ClientID = 1;
    bytes Payload = 2;
}

message ServerMsg {
    bytes Payload = 1;
}

message HandshakeMsg {
    bytes PublicKey = 1;
}

enum ErrorType {
    UNKNOWN = 0;
    HANDSHAKE_EXPIRED = 1;
}

message ErrorMsg {
    ErrorType type = 1;
}

message BaseMsg {
    oneof MsgType {
        ErrorMsg ErrorMsg = 1;
        HandshakeMsg HandshakeMsg = 2;
        ClientMsg ClientMsg = 3;
        ServerMsg ServerMsg = 4;
    }
}